### ISSUE TITLE
nixos-unstable: nixos-rebuild switch --upgrade fails with "This does not look like a tar archive"

### ISSUE THREAD

**Issue by rzetterberg** [NixOS/nixpkgs#8088]
### Problem description

I'm running NixOS unstable and was going to update today, so I ran: 

```bash
$ sudo nix-channel --update
$ sudo nixos-rebuild switch --upgrade
downloading Nix expressions from ‘https://nixos.org/releases/nixos/unstable/nixos-15.06pre62649.f93a8ee/nixexprs.tar.xz’...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
unpacking channels...
/nix/store/j7ghngyylfkl64xx1js6f2dfix48fw7g-xz-5.2.0/bin/xz: (stdin): File format not recognized
/nix/store/s16y2pk8lljsm37b55h2dsl6gpbsb62g-gnutar-1.27.1/bin/tar: This does not look like a tar archive
/nix/store/s16y2pk8lljsm37b55h2dsl6gpbsb62g-gnutar-1.27.1/bin/tar: Exiting with failure status due to previous errors
builder for ‘/nix/store/5df6zfj6qx8hcgqz4qhrqcagdgl5cw2s-nixos-15.06pre62649.f93a8ee.drv’ failed with exit code 2
error: build of ‘/nix/store/5df6zfj6qx8hcgqz4qhrqcagdgl5cw2s-nixos-15.06pre62649.f93a8ee.drv’ failed
cannot unpack the channels at /run/current-system/sw/bin/nix-channel line 150.
```

### What I have tried so far

I have been searching for other issues on github with similiar problem and found #7669. I tried the remove channel-cache workaround, but that didn't help. 

### My specifications

```bash
$ nixos-version 
15.05pre61665.4c01e6d (Dingo)
$ sudo nix-channel --list
nixos https://nixos.org/channels/nixos-unstable
$ nix-env --version
nix-env (Nix) 1.8
$ curl --version
curl 7.42.0 (x86_64-unknown-linux-gnu) libcurl/7.42.0 OpenSSL/1.0.1m zlib/1.2.8 libssh2/1.4.3
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: IPv6 Largefile NTLM NTLM_WB SSL libz TLS-SRP UnixSockets 
```

I can provide more information and testing upon request

**Comment by domenkozar** [comment]
Are you sure `sudo rm /nix/var/nix/channel-cache/*` didn't help?

**Comment by rzetterberg** [comment]
@domenkozar Here are the commands I ran: 

```bash
sudo rm -rf /nix/var/nix/channel-cache/
sudo nix-channel --update
sudo nixos-rebuild switch --upgrade
```

Now when I double checked, I ran these commands instead:

```bash
sudo rm -rf /nix/var/nix/channel-cache/
sudo nixos-rebuild switch --upgrade
```

And now it works! Looking at [the manual](https://nixos.org/nixos/manual/sec-upgrading.html) I now see that you shouldn't run the update like I did the first time. 

Sorry for the noise and thanks for the feedback!

**Comment by andrewrk** [comment]
I'm getting this error too.

```
$ nixos-version 
15.06pre64030.e1af50c (Dingo)
$ sudo nix-channel --list
nixos https://nixos.org/channels/nixos-unstable
$ nix-env --version
nix-env (Nix) 1.9pre4153_b64988b
$ curl --version
curl 7.42.1 (x86_64-unknown-linux-gnu) libcurl/7.42.1 OpenSSL/1.0.1m zlib/1.2.8 libssh2/1.5.0
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: IPv6 Largefile NTLM NTLM_WB SSL libz TLS-SRP UnixSockets 
```

**Comment by edolstra** [comment]
@andrewrk This should be fixed in Nix 1.9.
